Pylint and continuous integration

Saityi opened the pull request:
@sourlows test running thingy and pylint running thingy

Saityi committed:
Command line test running script

Saityi committed:
moved test_runner to root directory and changed pathnames in it to be relative to that.

Saityi committed:
pylint running script and pylintrc for test and src/app added; run src tests with -s, test tests with -t

Saityi committed:
.

Saityi committed:
Update pylint_runner.py

Saityi committed:
This file is mirrored in api/base.py and did not appear to have any use. Nothing uses it or inherits from it.

Saityi committed:
Merge remote-tracking branch 'origin/pylint_and_continuous_integration' into pylint_and_continuous_integration

Saityi committed:
Cleared up issues in src/app exposed by pylint; missing docstrings still need to be implemented. Also: two unused arguments, two argument numbers differ from overridden build_key method to be fixed.

Saityi committed:
Cleared up issues in tests exposed by pylint; everything but missing docstrings fixed.

sourlows commented on src/pylintrc:
```diff
@@ -0,0 +1,53 @@
+reports=n
+include-ids=y
+[MASTER]
+# Can safely ignore libs - VBC doesn't manage the code that lives in these dirs.
```
delete this comment

sourlows commented on test/pylintrc:
```diff
@@ -0,0 +1,54 @@
+reports=n
+include-ids=y
+[MASTER]
+# Can safely ignore libs - VBC doesn't manage the code that lives in these dirs.
```
delete this comment

sourlows commented:
can you explain how to run these in README.md ? Delete those comments and merge this in when you're ready. Good job!

Saityi committed:
Updated pylint_runner.py to run on all source code, not just that in src/app;
Fixed issues exposed in main.py by pylint analysis;
Updated an issue exposed in the league_test by pylint changes;
Added information to README.md about how to use the test runner and pylint analysis scripts

Saityi commented on test/api/league_test.py:
```diff
@@ -89,22 +88,28 @@ def test_post_returns_400_if_invalid_rating_scheme_is_specified(self):
     def test_post_new_league_when_created_successfully(self):
-        self.create_test_league()
-        self.mock_function_in_setup('app.api.league.create_league', return_value=self.league)
+        self.create_test_user()
```
@sourlows Mocking create_league() for this test bugged out when I changed stuff in src/main.py (??). I'm iffy on the way I changed it, but I couldn't find a way to make it correctly test post as written before.

Saityi committed:
Update README.md

Fixing formatting

Saityi committed:
Update README.md

'

Saityi committed:
Made pylint_runner's --help to run by default if no valid arguments are used.

Saityi committed:
Suppressing pylint warnings
